### PR TITLE
WIP: patch-style imports

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
Interesting bits as far as the implementation are here: https://github.com/pulumi/pulumi/pull/4505/files#diff-674141a8d712fdb099f5fa5e79270d6c

Here's a tech demo: https://asciinema.org/a/HL7cb2geC7B0rVu81qaZE4iJV

File: pkg/resource/deploy/source_eval.go [hunk 1]
```diff
@@ -300,7 +300,7 @@ func (d *defaultProviders) newRegisterDefaultProviderEvent(
 	event := &registerResourceEvent{
 		goal: resource.NewGoal(
 			providers.MakeProviderType(req.Package()),
-			req.Name(), true, inputs, "", false, nil, "", nil, nil, nil, nil, nil, nil, "", nil),
+			req.Name(), true, inputs, "", false, nil, "", nil, nil, nil, nil, nil, nil, "", nil, false),
```
Comment (lblackstone) [inline pkg/resource/deploy/source_eval.go:303]: Would it be worth wrapping these params in a struct? This list is getting unwieldy.

Comment (pgavlin) [reply pkg/resource/deploy/source_eval.go:303]: I was thinking about that, too, esp. given that so many are essentially optional. Ironically, all this method does is... build a struct. So we could just use it directly and get rid of these incomprehensible calls.

**Comment by pgavlin** [comment]
Closing this--we're not moving forward here.

**Pull request closed without merging**